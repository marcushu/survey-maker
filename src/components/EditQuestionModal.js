import React, { useEffect, useState } from 'react';
import Modal from 'react-bootstrap/Modal';
import Binary from './surveyMakerQuestions/Binary';
import Text from './surveyMakerQuestions/Text';
import MultipleChoice from './surveyMakerQuestions/MultipleChoice';
import PropTypes from 'prop-types';


export default function EditQuestionModal({ questionEntry, show, hideModal, updateValues }) {
  const [currentComponent, setcurrentComponent] = useState(null);

  useEffect(() => {
    if (questionEntry) {
      switch (questionEntry.questionType) {
        case "BINARY":
          setcurrentComponent(<Binary addQuestion={updateValues} />);
          break;
        case "TEXT":
          setcurrentComponent(<Text addQuestion={updateValues} />);
          break;
        case "MULTCHOICE":
          setcurrentComponent(<MultipleChoice addQuestion={updateValues} />);
          break;
        default:
          break;
      }
    }

  }, [questionEntry, updateValues]);
  

  return (
    <Modal show={show} onHide={hideModal}>
      <Modal.Header closeButton>
        <Modal.Title>Edit Question</Modal.Title>
      </Modal.Header>
      <Modal.Body className="text-center" style={{backgroundColor: "rgb(218, 218, 218)"}}>
        {currentComponent}
      </Modal.Body>
    </Modal>
  )
}

EditQuestionModal.propTypes = {
  questionEntry: PropTypes.string,
  show: PropTypes.bool,
  hideModal: PropTypes.func,
  updateValues: PropTypes.func
}