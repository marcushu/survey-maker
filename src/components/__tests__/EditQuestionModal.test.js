import React from 'react';
import { render } from '@testing-library/react';
import EditQuestionModal from '../EditQuestionModal';

describe("editQuestionModal", () => {
    const mockFunc = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <EditQuestionModal
            questionEntry={"mock Question"}
            show={true}
            hideModal={mockFunc}
            updateValues={mockFunc} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
    
})