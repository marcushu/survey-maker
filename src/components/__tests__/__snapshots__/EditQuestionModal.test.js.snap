// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editQuestionModal renders 1`] = `<DocumentFragment />`;
